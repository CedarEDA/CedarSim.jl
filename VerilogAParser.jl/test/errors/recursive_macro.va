`define MPIcc(nam,def,uni,lwr,upr,des) (*units=uni, desc=des*) parameter integer nam=def from[lwr:upr] ;
`define MPRco(nam,def,uni,lwr,upr,des) (*units=uni, desc=des*) parameter real nam=def from[lwr:upr) ;
`define MPRcc(nam,def,uni,lwr,upr,des) (*units=uni, desc=des*) parameter real nam=def from[lwr:upr] ;
`define MPRnb(nam,def,uni,        des) (* units=uni, desc=des *) parameter real    nam=def;
`define IPRnb(nam,def,uni,        des) (* units=uni, type="instance", desc=des *) parameter real    nam=def;
`define IPIco(nam,def,uni,lwr,upr,des) (*units=uni, type="instance", desc=des*) parameter integer nam=def from[lwr:upr) ;
`define IPRco(nam,def,uni,lwr,upr,des) (*units=uni, type="instance", desc=des*) parameter real nam=def from[lwr:upr) ;
`define IMPIcc(nam,def,uni,lwr,upr,des) (*units=uni, type="both", desc=des*) parameter integer nam=def from[lwr:upr] ;
`define IMPRnb(nam,def,uni,       des) (*units=uni, type="both", desc=des*) parameter real nam=def;
`define Obs(name, des) (* desc = des *) real name;

// TODO: Units and desc
`define ScalePars(nam) \
    parameter real L``nam=0.0; \
    parameter real W``nam=0.0; \
    parameter real P``nam=0.0;

`define MPRnbS(nam,def,uni,        des) `MPRnbS(nam,def,uni,        des) `ScalePars(nam)
`define MPRcoS(nam,def,uni,lwr,upr,des) `MPRcoS(nam,def,uni,lwr,upr,des) `ScalePars(nam)

module recursive_macro(x);
    `MPRnbS( AIGSD         , TYPE == `ntype ? 1.36e-2 : 9.8e-3, "√(Fs²/g)m⁻¹"       , "Parameter for Igs,d")
endmodule