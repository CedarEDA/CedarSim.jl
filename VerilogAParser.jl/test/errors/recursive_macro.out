[31m[1m╭ ERROR [VAS04]:[22m[39m[31m Macro Evaluation recursed.[39m
[31m│[39m[90m   ╭───── recursive_macro.va:22[39m
[31m│[90m22[0m [90m│[0m     `MPRnbS( AIGSD         , TYPE == `ntype ? 1.36e-2 : 9.8e-3, "√(Fs²/g)m⁻¹"       , "Parameter for Igs,d")
[31m│[39m[90m   ├───── Using macro [39m[92m`MPRnbS[39m[90m at recursive_macro.va:18[39m
[31m│[90m18[0m [90m│[0m `define MPRnbS(nam,def,uni,        des) `MPRnbS(nam,def,uni,        des) `ScalePars(nam)
[31m│[39m[90m   ├───── Expanded to[39m
[31m│[90m22[0m [90m│[0m [92m [91m`MPRnbS[92m(
[31m│[0m   [90m┊[0m  [91m━━━┳━━━
[31m│[0m   [90m┊[0m [91m✗ Recusive macro expansion[0m
[31m│[39m[90m   │[39m
[31m│[39m[90m   ╰───── [39mMacro Evaluation recursed [39m[1m`MPRnbS[22m.
[31m│[39m
[31m│[39m[34m help: [39mThe code made use of the macro [39m[1m`MPRnbS[22m while evaluation the same macro.
[31m│[39m       Check for spelling mistakes or definition ordering.
[31m╯[39m
